src/components/contact-govuk-one-login/contact-govuk-one-login-controller.ts

```diff
@@ -1,7 +1,30 @@
 import { Request, Response } from "express";
+import { isValidUrl } from "./../../utils/strings";
+
+const CONTACT_ONE_LOGIN_TEMPLATE = "contact-govuk-one-login/index.njk";
+const NOT_FOUND_TEMPLATE = "common/errors/404.njk";
 
 export function contactGet(req: Request, res: Response): void {
   const data = {};
-
-  res.render("contact-govuk-one-login/index.njk", data);
+  const fromUrl = getFromUrlAndSaveIt(req);
+  if (fromUrl) {
+    res.render(CONTACT_ONE_LOGIN_TEMPLATE, data);
+  } else {
+    req.log.error(
+      `request and session contain no valid fromUrl`
+    );
+    res.render(NOT_FOUND_TEMPLATE);
```

Comment:
I'd appreciate your thoughts on the best approach here.

Reply:
Hmm good point. 

This is just a web page on the internet, so it's entirely possible people end up on it without a URL param (eg. via a Google search). I think we still need to let those people contact us, so if there's no `fromUrl` in the params or in the session then we should not include that param in the link off to the contact form. I'll check with Auth that they can still handle that case.